pkg/providers/tinkerbell/hardware/etcd.go

```diff
@@ -82,3 +78,19 @@ func (er *ETCDReader) getUnprovisionedTinkerbellHardware(ctx context.Context) ([
 
 	return selectedHardware.Items, nil
 }
+
+// RufioMachinesFromEtcd fetches rufio machine objects from the cluster and inserts into catalogue.
+func (er *ETCDReader) RufioMachinesFromEtcd(ctx context.Context) error {
+	var rufioMachines rufiov1alpha1.MachineList
+	if err := er.client.List(ctx, &rufioMachines, &client.ListOptions{Namespace: constants.EksaSystemNamespace}); err != nil {
+		return fmt.Errorf("listing rufio machines: %v", err)
+	}
+
+	for i := range rufioMachines.Items {
+		if err := er.catalogue.InsertBMC(&rufioMachines.Items[i]); err != nil {
+			return err
+		}
```

Comment:
Does this work?
```suggestion
	for _,  rufioMachine := range rufioMachines.Items {
		if err := er.catalogue.InsertBMC(&rufioMachine); err != nil {
			return err
		}
```

Reply:
Actually this caused an issue. Since we are using pointers InsertBMC inserts the same rufioMachine pointer each time. Thereby each object inserted was always the last one in the list.

Reply:
oh wait that's how it works? So `InsertBMC()` takes a pointer to the BMC and not the BMC object itself?

Reply:
Yah it takes a pointer. It was hard to notice but when I tested it caused issues and I needed to change it back.